### ISSUE TITLE
GPU memory not being released after using demos/musicgen_app.py to complete a single inference

### ISSUE THREAD

**Issue by OroChippw** [facebookresearch/audiocraft#321]
![image](https://github.com/facebookresearch/audiocraft/assets/75830178/18b3dc96-87ad-4e9e-b44c-c4b492279ea2)
It's a pity that I only saw such a powerful model. Thank you very much for your contribution.But I found that after I used demos/musicgen_app.py to complete the text audio generation inference, my GPU memory was not released, and an OOM error occurred when I performed inference again. I had to close my service and reopen it, and proceed again. A reasoning. Is there something wrong with my operation?Think you~😊

**Comment by 0xlws** [comment]
hey, i dont think there is anything wrong with your operation, please see:

#### tldr: 16gb recommended

https://github.com/facebookresearch/audiocraft/blob/main/docs/MUSICGEN.md#installation
and
https://github.com/facebookresearch/audiocraft/blob/main/docs/MUSICGEN.md#api

the model is being stored in memory which allows it to be used multiple times without having to unload/reload again (which can be time consuming the larger the model gets).
as suggested in the readme you can try shorter audio lengths, or look for one of the many forks which allow for longer generations
https://github.com/camenduru/MusicGen-colab
(cant recommend any as i havent used them myself)
hope it helps

**Comment by OroChippw** [comment]
Then it seems that I need to use better equipment to complete my task. Thank you for your answer and I look forward to your greater masterpieces.😊
